src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -1251,18 +1259,28 @@ public function process_posts( $data = [], $start_immediately = false ) {
 			$service->confirm_import( $this->meta );
 		}
 
+		// CSV should be processed right away as does not have support for batch pushing.
+		$is_not_csv = empty( $data ) || empty( $data['origin'] ) || $data['origin'] !== 'csv';
 		// if this is a batch push record then set its queue to fetching
 		// to feed the UI something coherent
-		if ( ! $start_immediately && ! $this->is_polling() ) {
+		if ( $is_not_csv && ! $this->is_polling() ) {
 			// @todo let's revisit this to return when more UI is exposed
-			$queue = new Tribe__Events__Aggregator__Record__Queue( $this, 'fetch' );
+			$queue = new Tribe__Events__Aggregator__Record__Batch_Queue( $this );
+
+			if ( $start_immediately ) {
+				$queue->process();
+
+				return $queue->activity();
+			}
 
 			return $queue;
 		}
 
 		$items = $this->prep_import_data( $data );
 
 		if ( is_wp_error( $items ) ) {
+			tribe( 'logger' )->log_debug( "Error while preparing the items of the request.", 'EA Process Posts.' );
```

Comment:
:no_entry_sign: String "Error while preparing the items of the request." does not require double quotes; use single quotes instead

<sub>hash: d9b7f0c171d4c2f2db359e13aec04882</sub>